javaparser-visitor-generator/src/main/java/com/github/javaparser/generator/visitor/VisitorGenerator.java

```diff
@@ -0,0 +1,85 @@
+package com.github.javaparser.generator.visitor;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.generator.utils.SourceRoot;
+import com.github.javaparser.metamodel.BaseNodeMetaModel;
+import com.github.javaparser.metamodel.JavaParserMetaModel;
+import com.github.javaparser.metamodel.PropertyMetaModel;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A class that makes it easier to generate visitor classes.
+ * It will create missing visit methods on the fly,
+ * and will ask you to fill in the bodies of the visit methods.
+ */
+public abstract class VisitorGenerator {
+    protected final JavaParser javaParser;
+    protected final SourceRoot sourceRoot;
+    private final String pkg;
+    private final String visitorClassName;
+    private final String returnType;
+    private final String argumentType;
+    private final boolean createMissingVisitMethods;
+    protected final JavaParserMetaModel javaParserMetaModel;
+
+    public VisitorGenerator(JavaParser javaParser, SourceRoot sourceRoot, String pkg, String visitorClassName, String returnType, String argumentType, boolean createMissingVisitMethods, JavaParserMetaModel javaParserMetaModel) {
+        this.javaParser = javaParser;
+        this.sourceRoot = sourceRoot;
+        this.pkg = pkg;
+        this.visitorClassName = visitorClassName;
+        this.returnType = returnType;
+        this.argumentType = argumentType;
+        this.createMissingVisitMethods = createMissingVisitMethods;
+        this.javaParserMetaModel = javaParserMetaModel;
+    }
+
+    public void generate() throws IOException {
```

Comment:
No way, I don't like sonar that much anyway ... :), my focus in my masterthesis (master overall) is code quality and bad smells ;)